packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -217,13 +223,17 @@ class Dart2JSTarget extends Target {
     }
 
     final String? dart2jsOptimization = environment.defines[kDart2jsOptimization];
+    final bool dumpInfo = environment.defines[kDart2jsDumpInfo] == 'true';
+    final bool noFrequencyBasedMinification = environment.defines[kDart2jsNoFrequencyBasedMinification] == 'true';
     final File outputJSFile = environment.buildDir.childFile('main.dart.js');
     final bool csp = environment.defines[kCspMode] == 'true';
 
     final ProcessResult javaScriptResult = await environment.processManager.run(<String>[
       ...sharedCommandOptions,
       if (dart2jsOptimization != null) '-$dart2jsOptimization' else '-O4',
       if (buildMode == BuildMode.profile) '--no-minify',
+      if (dumpInfo) '--dump-info',
```

Comment:
where does the dump-info go?

Reply:
because this may end up in a build subdirectory that is difficult to find by default.

Reply:
it *does* go in the build subdirectory. The idea is: if a user wants it, they can go look for it.

Again, I expect this to be a feature for advanced users.

Reply:
sgtm